File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteMiddle = function(head) {
    if (head==null || head.next==null){
        return null
    }
    if (head.next.next==null){
        head.next =null;
        return head
    }
    //     if (head.next.next.next==null){
    //     head.next =head.next.next
    //     return head
    // }
    let slow = head;
    let  fast = head;
    let prevslow = null;
    while(fast != null && fast.next!=null){
        fast = fast.next.next;
        prevslow= slow;
        slow= slow.next;
    }
   
    prevslow.next = slow.next;
    return  head;
    
};

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        ListNode* slow = head;
        ListNode* fast = head;
        ListNode* prevslow = NULL;
        while (fast && fast->next) {
            prevslow = slow;
            slow = slow->next;
            fast = fast->next->next;
        }
        if (prevslow == NULL)
            return head->next;
        prevslow->next = slow->next;
        return head;
    }

}
;